### PR TITLE
`PlacementMode` -> `Placement`

### LINKED ISSUE #9583: Flyouts should share PlacementMode with popups
This includes adding `PlacementRect` and `AnchorAndGravity` mode support for Flyouts with the corresponding properties. We can add existing FlyoutPlacementMode members to our main PlacementMode enum.

### PR DESCRIPTION
**Pull request by robloo**
## What does the pull request do?

Follow-up to https://github.com/AvaloniaUI/Avalonia/pull/10492 as discussed starting here: https://github.com/AvaloniaUI/Avalonia/pull/10492#issuecomment-1459449102

## What is the current behavior?

 * `PlacementMode` and `Placement` are used interchangeably in a non-standardized way
 * `PlacementMode` doesn't match WPF property naming
   * [Popup.Placement Property](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.popup.placement?view=windowsdesktop-7.0)
   * [ContextMenu.Placement Property](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.contextmenu.placement?view=windowsdesktop-7.0)

## What is the updated/expected behavior with this PR?

All properties standardized to use `Placement` naming.

## How was the solution implemented (if it's not obvious)?

Obvious.

## Checklist

- ~[ ] Added unit tests (if possible)?~
- [x] Added XML documentation to any related classes?
- ~[ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation~

## Breaking changes

Yes, property name change:

 1. Popup.PlacementMode to Popup.Placement
 1. ContextMenu.PlacementMode to ContextMenu.Placement

## Obsoletions / Deprecations

None

## Fixed issues

Also fixes https://github.com/AvaloniaUI/Avalonia/issues/9583

### CODE REVIEW

File: src/Avalonia.Controls/ContextMenu.cs [hunk 1]
```diff
@@ -108,99 +108,72 @@ public ContextMenu(IMenuInteractionHandler interactionHandler)
...
+            get => GetValue(PlacementGravityProperty);
+            set => SetValue(PlacementGravityProperty, value);
         }
 
-        /// <summary>
-        /// Gets or sets the placement mode of the context menu in relation to the<see cref="PlacementTarget"/>.
-        /// </summary>
-        public PlacementMode PlacementMode
+        /// <inheritdoc cref="Popup.Placement"/>
+        public PlacementMode Placement
```
Comment (robloo) [inline src/Avalonia.Controls/ContextMenu.cs:153]: All of the Popup "Placement" properties shouldn't really be duplicated here. There is an open question if ContextMenu should be deriving from PopupFlyoutBase as well (like MenuFlyout).

File: src/Avalonia.Controls/ContextMenu.cs [hunk 2]
```diff
@@ -108,99 +108,72 @@ public ContextMenu(IMenuInteractionHandler interactionHandler)
...
+            PlacementProperty.OverrideDefaultValue<ContextMenu>(PlacementMode.Pointer);
             ContextMenuProperty.Changed.Subscribe(ContextMenuChanged);
             AutomationProperties.AccessibilityViewProperty.OverrideDefaultValue<ContextMenu>(AccessibilityView.Control);
             AutomationProperties.ControlTypeOverrideProperty.OverrideDefaultValue<ContextMenu>(AutomationControlType.Menu);
         }
 
-        /// <summary>
-        /// Gets or sets the Horizontal offset of the context menu in relation to the <see cref="PlacementTarget"/>.
-        /// </summary>
+        /// <inheritdoc cref="Popup.HorizontalOffset"/>
```
Comment (robloo) [inline src/Avalonia.Controls/ContextMenu.cs:117]: I'm inheriting the docs from Popup now. The only downside is "context menu" wording is replaced by "popup". I don't consider this a major issue and it's better to find more general wording in the base docs than duplicate them everywhere.

File: src/Avalonia.Controls/Primitives/Popup.cs [hunk 3]
```diff
@@ -256,8 +256,8 @@ public PlacementMode PlacementMode
         /// </remarks>
         public Rect? PlacementRect
```
Comment (robloo) [inline src/Avalonia.Controls/Primitives/Popup.cs:257]: `PlacementRect` seems like an unnecessary deviation from naming conventions and WPF which uses `PlacementRectangle`. Honestly, I would like to switch to WPF terminology: https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.popup.placementrectangle?view=windowsdesktop-7.0

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0031879-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by kekekeks** [comment]
Can we have `[Obsolete]` `PlacementMode` and `PlacementModeProperty`? I'd like to not cause compilation errors unless necessary.

**Comment by robloo** [comment]
Really isn't ideal to support considering all the other breaking changes. I can't think of many cases where two styled properties exist that do the same thing either.

That said if it's required it can be done.

**Comment by kekekeks** [comment]
There won't be two separate properties, just an alias: `public static StyledProperty<PlacementMode> PlacementModeProperty = PlacementProperty;`

**Comment by robloo** [comment]
@kekekeks Updated to add back an obsolete alias for PlacementMode

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0031912-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Controls/ContextMenu.cs [hunk 4]
```diff
@@ -57,8 +57,18 @@ public class ContextMenu : MenuBase, ISetterValue, IPopupHostProvider
         /// <summary>
         /// Defines the <see cref="PlacementMode"/> property.
         /// </summary>
-        public static readonly StyledProperty<PlacementMode> PlacementModeProperty =
-            Popup.PlacementModeProperty.AddOwner<ContextMenu>();
+        [Obsolete("Use the Placement property instead.")]
+        public static readonly StyledProperty<PlacementMode> PlacementModeProperty = PlacementProperty!;
```
Comment (jp2masa) [inline src/Avalonia.Controls/ContextMenu.cs:61]: Isn't it actually null here? `PlacementProperty` is defined below, so I think it's initialized after this one.

Comment (robloo) [reply src/Avalonia.Controls/ContextMenu.cs:61]: Lol, It was late and I didn't think...

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0031914-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

File: src/Avalonia.Controls/Primitives/Popup.cs (hunk 3 above)
Comment (robloo) [reply src/Avalonia.Controls/Primitives/Popup.cs:271]: I guess there is precedent for the new naming: https://learn.microsoft.com/en-us/uwp/api/windows.ui.input.pointerpointproperties.contactrect?view=winrt-22621

`PointerPointProperties.ContactRect`

**Pull request merged**